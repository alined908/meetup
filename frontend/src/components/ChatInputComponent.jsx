import React, {Component} from 'react'
import {connect} from 'react-redux';
import {setTypingValue, addMessage} from "../actions/chat";


class ChatInputComponent extends Component{
    handleChange = (e) => {
        console.log(e.target.value);
        this.props.setTypingValue(e.target.value);        
    }

    handleSubmit = (e) => {
        if (e.key === "Enter"){
            e.preventDefault();
            this.props.addMessage(e.target.value, this.props.room)
        }
    }

    render(){
        return (
            <form className="chat-input-form">
                <input className="chat-input"
                    type="text"
                    onChange={this.handleChange} 
                    onKeyPress={this.handleSubmit}
                    value={this.props.message} 
                    placeholder="Type here to send a message">
                </input>
            </form>
        )
    } 
}

function mapStateToProps(state){
    return {
        message: state.chat.setTypingValue,
        room: state.chat.activeRoom
    }
}

const mapDispatchToProps = {
    setTypingValue,
    addMessage
}

export default connect(mapStateToProps, mapDispatchToProps)(ChatInputComponent)