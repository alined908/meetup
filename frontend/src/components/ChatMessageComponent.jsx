import React, {Component} from 'react'
import moment from 'moment';

class ChatMessageComponent extends Component {
    render (){
        const [id, message, timestamp, is_read, room, sender, receipient] = Object.values(this.props.message)
        console.log(this.props.user)
        let user = JSON.parse(this.props.user);

        var is_user_msg = user.id === sender
    
        if (is_user_msg){
            var name = user.first_name
        } else {
            var name = this.props.members[sender].first_name
        }

        return (
            <div className={`chat-msg-wrapper ${is_user_msg ? "is-user": ""}`}>
                <div className={`chat-msg-info ${is_user_msg ? "is-user": ""}`}>
                    <div className="chat-msg-id">
                        {name}
                    </div>
                    <div className="chat-msg-time">
                        {moment(timestamp).local().format("MMM DD hh:mm A")}
                    </div>
                </div>
                <div className={`chat-msg ${is_user_msg ? "is-user": ""}`}>{message}</div>
            </div>
        )
    }
}


export default ChatMessageComponent