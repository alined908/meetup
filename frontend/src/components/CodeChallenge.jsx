import React, {Component} from 'react'
import {axiosClient} from '../accounts/axiosClient'
import styles from "../styles/challenge.module.css"
import {DragDropContext, Droppable, Draggable} from 'react-beautiful-dnd'
import {CircularProgress} from '@material-ui/core'
import CloseIcon from '@material-ui/icons/Close';

const copy = (filled, destination, numCategories) => {
    const col = Number(destination.droppableId.split("-")[1]) % numCategories
    return Object.assign([], filled, {[col]: true})
}

const move = (filled, source, destination, numCategories) => {
    const oldCol = Number(source.droppableId.split("-")[1]) % numCategories
    const newCol = Number(destination.droppableId.split("-")[1]) % numCategories
    return Object.assign([], filled, {[oldCol]: false, [newCol]: true})
}   

const getStyle = (style, snapshot) => {
    if (!snapshot.isDragging) return {};
    if (!snapshot.isDropAnimating) {
      return style;
    }
    return {
      ...style,
      transitionDuration: `0.001s`,
    };
  }
  
class Card extends Component {
    constructor(props){
        super(props);
        this.state = {
            isHovered: false
        }
    }

    render () {
        const restaurant = this.props.restaurant
        return (
            <div className={`${styles.rstCard} ${this.state.isHovered ? styles.closeHovered : ""}`}>
                {!this.props.isMenu && 
                    <CloseIcon
                        size="small" 
                        color="secondary"
                        className={styles.closeIcon}
                        onClick={this.props.handleDelete} 
                        onMouseEnter={() => this.setState({isHovered: true})} 
                        onMouseLeave={() => this.setState({isHovered: false})}
                    />
                }
                <div className={styles.rstCardTop}>
                    <img className={styles.rstImg} src={restaurant.yelp_image} />
                </div>
                <div className={styles.rstCardBottom}>
                    <span>{restaurant.name}</span>
                    <div className={styles.rstCardCategories}>
                        {this.props.isMenu && this.props.categories.map((category, index) => 
                            this.props.filled[index] && <img
                                key={category.id} 
                                className={styles.categoryPicture} 
                                src={`${process.env.REACT_APP_S3_STATIC_URL}${category.api_label}.png`}
                            />
                        )}
                    </div>
                </div>
            </div>
        )
    }
}

class Restaurant extends Component {

    render () {
        const restaurant = this.props.restaurant

        return (
            <Draggable draggableId={this.props.isMenu ? `restaurant-${restaurant.id}`: `category-${this.props.index}`} index={this.props.index}>
                {(provided, snapshot) => (
                    <React.Fragment>
                        <div 
                            ref={provided.innerRef} 
                            {...provided.draggableProps}
                            {...provided.dragHandleProps}
                            className={styles.restaurant}
                            style={getStyle(provided.draggableProps.style, snapshot)}
                        >   
                            <Card 
                                restaurant={restaurant} 
                                isMenu={this.props.isMenu}
                                handleDelete={this.props.handleDelete}
                                filled={this.props.filled}
                                categories={this.props.categories}
                            />
                        </div>
                        {snapshot.isDragging && this.props.isMenu && (
                            <div className={styles.restaurant}>
                            
                            </div>
                        )}
                    </React.Fragment>
                )}
                
            </Draggable>
        )
    }
}

class RestaurantRow extends Component {
    
    constructor(props){
        super(props);
        this.state = {
            filled: new Array(props.categories.length).fill(false)
        }
    }

    handleDelete = (index) => {
        const newFilled = Object.assign([], this.state.filled, {[index]: false})
        this.setState({filled: newFilled})
    }

    onDragEnd = result => {
        const {source, destination} = result;

        if (!destination){
            return;
        }  

        const numCategories = this.props.categories.length;
        let filled;

        if (source.droppableId.includes("restaurant")) {
            filled = copy(this.state.filled, destination, numCategories)
        } else if (source.droppableId.includes("category")){
            filled = move(this.state.filled, source, destination, numCategories)
        }
        
        this.setState({filled})
    }

    render () {
        const restaurant = this.props.restaurant
        const categories = this.props.categories
        const offset = restaurant.id * categories.length

        return (
            <DragDropContext onDragEnd={this.onDragEnd}>
                <div className={styles.restaurantRow}>
                    <Droppable droppableId={`restaurant-${restaurant.id}`} isDropDisabled={true}>
                        {(provided, snapshot) => (
                            <div ref={provided.innerRef} {...provided.droppableProps} className={styles.restaurantWrapper}>
                                <Restaurant 
                                    isMenu={true}
                                    filled={this.state.filled}
                                    restaurant={restaurant}
                                    categories={categories}
                                    index={this.props.rstIndex}
                                />
                                <span style={{display: "none"}}>
                                    {provided.placeholder}
                                </span>
                            </div>
                        )}
                    </Droppable>
                    <div className={styles.columns}>
                        {categories.map((category, index) => 
                            <Droppable key={category.id} droppableId={`category-${offset + index}`}>
                                {(provided, snapshot) => (
                                    <div ref={provided.innerRef} className={styles.block}>
                                        {this.state.filled[index] ?  
                                            <Restaurant 
                                                isMenu={false}
                                                restaurant={restaurant}
                                                handleDelete={() => this.handleDelete(index)}
                                                index={offset + index} />
                                            :   
                                            <div className={styles.category}/>
                                        }
                                        <span style={{display: "none"}}>
                                            {provided.placeholder}
                                        </span>
                                    </div>  
                                )}
                            </Droppable>
                        )}
                    </div>
                </div>
            </DragDropContext>
        )
    }
}

class CodeChallenge extends Component {

    constructor(){
        super();
        this.state = {
            restaurants: [],
            categories: [],
            isFetching: true,
            isInitialized: false
        }
    }

    async componentDidMount() {

        const [restaurantResponse, categoryResponse] = await Promise.all([
            axiosClient.get(`/api/restaurants/challenge/`),
            axiosClient.get(`/api/categories/challenge/`)
        ])
        this.setState({
            restaurants: restaurantResponse.data, 
            categories: categoryResponse.data,
            isFetching: false,
            isInitialized: true
        })
    }

    render () {
        return (
            <>
                {this.state.isInitialized ?
                    <div className={`${styles.outerWrapper}`}>
                        <div className={styles.topWrapper}>
                            <div className={styles.restaurantsWrapper}>
                                <div className={styles.header}>
                                    Restaurants
                                </div>
                                <div className={styles.restaurantIndent}/>
                            </div>
                            <div className={styles.categoriesWrapper}>
                                <div className={styles.header}>
                                    Categories
                                </div>
                                <div className={styles.categories}>
                                    {this.state.categories.map((category, index) => 
                                        <div key={index} className={styles.categoryHeaderWrapper}>
                                            <div className={styles.categoryHeader}>
                                                <img 
                                                    className={styles.categoryPicture} 
                                                    src={`${process.env.REACT_APP_S3_STATIC_URL}${category.api_label}.png`}
                                                />
                                                {category.label}
                                            </div>
                                        </div>
                                    )}
                                </div>
                            </div>
                        </div>
                        <div className={styles.bottomWrapper}>
                            <div className={styles.restaurants}>
                                {this.state.restaurants.map((restaurant, index) =>
                                    <RestaurantRow 
                                        key={restaurant.id}
                                        rstIndex={index}
                                        restaurant={restaurant} 
                                        categories={this.state.categories}
                                    />
                                )}
                            </div>
                        </div>
                    </div>
                    :
                    <div className="loading">
                        <CircularProgress size={30}/>
                    </div>
                    
                }       
            </>
        )
    }
}

export default CodeChallenge