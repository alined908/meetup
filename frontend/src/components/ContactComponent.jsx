import React, {Component} from 'react'
import {connect} from 'react-redux';
import {setActiveRoom} from '../actions/chat';

class ContactComponent extends Component {
    handleClick = (uri) => {
        console.log("handle click")
        this.props.setActiveRoom(uri);
    }
    
    render (){
        const [id, uri, name, timestamp, members] = this.props.room

        return (
            <div className="chat-contact" onClick={() => this.handleClick(uri)}>
                <div>{name}</div>
                <div>{Object.keys(members).length + " Members"}</div>
            </div>
        )
    }
}
const mapDispatchToProps = {
    setActiveRoom
}

export default connect(null, mapDispatchToProps)(ContactComponent)