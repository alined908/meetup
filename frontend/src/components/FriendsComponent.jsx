import React, {Component} from 'react'
import {getFriends, addFriend, deleteFriend} from "../actions/friend"
import {reduxForm, Field} from 'redux-form';
import {connect} from 'react-redux'
import {Button} from '@material-ui/core';
import {compose} from 'redux'

class FriendsComponent extends Component{

    componentDidMount(){
        this.props.getFriends();
    }

    onSubmit = (formProps) => {
        this.props.addFriend(formProps)
    }

    render(){
        const {handleSubmit} = this.props;

        const addFriendForm = () => {
            return (<form onSubmit={handleSubmit(this.onSubmit.bind(this))}>
                <div>{this.props.errorMessage}</div>
                <Field name="email" component="input" label="email"/>
                <Button type="submit" variant="contained" color="primary">Add Friend</Button>
            </form>)
        }

        return (
            <div>
                {!this.props.isFriendsInitialized && <div>...Initializing Friends</div>}
                {this.props.isFriendsInitialized && this.props.friends.map((friendship) => 
                    friendship.user.email + ", "
                )}
                {this.props.isFriendsInitialized && addFriendForm()}
            </div>
        )
    }
}

function mapStateToProps(state){
    return {
        friends: state.user.friends,
        isFriendsInitialized: state.user.isFriendsInitialized,
        errorMessage: state.user.errorMessage
    }
}

const mapDispatchToProps = {
    getFriends,
    addFriend,
    deleteFriend
}

export default compose(connect(mapStateToProps, mapDispatchToProps), reduxForm({form: 'friend'}))(FriendsComponent)