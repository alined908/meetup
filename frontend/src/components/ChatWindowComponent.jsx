import React, {Component} from 'react'
import ChatMessageComponent from "./ChatMessageComponent"
import ChatInputComponent from "./ChatInputComponent"
import {connect} from 'react-redux'

class ChatWindowComponent extends Component {

    componentDidMount() {
        console.log("mounted chat window")
        this.scrollToBottom();
      }
    componentDidUpdate() {
        console.log("updated chat window")
        this.scrollToBottom();
    }
    scrollToBottom = () => {
        console.log("scroll to bottom called")
        this.messagesEnd && this.messagesEnd.scrollIntoView({ behavior: "smooth" });
    };
    
    render () {
        return (
            <div className="chat-window" ref={this.chatsRef}>
                <div className="chat-header">
                    Conversation for uri: {this.props.activeRoom}
                </div>
                <div className="chat-messages">
                    {this.props.messages && this.props.messages.map((msg) => <ChatMessageComponent user={this.props.user} message={msg.message} members={this.props.activeChatMembers}/>)}
                </div>
                {this.props.activeChatMembers && <div ref={(el) => { this.messagesEnd = el; }} className="chat-input">
                    <ChatInputComponent/>
                </div>}
            </div>
        )
    }
}

function mapStateToProps(state){
    if (state.chat.activeRoom in state.chat.rooms){
        return {
            activeChatMembers: state.chat.rooms[state.chat.activeRoom].members,
            user: state.auth.user
        }
    } else {
        return {
            activeChatMembers: null,
            user: state.auth.user
        }
    }
    
}


export default connect(mapStateToProps)(ChatWindowComponent)