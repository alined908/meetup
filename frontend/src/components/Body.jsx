import React, {Component} from 'react';
import {Route, Switch} from 'react-router-dom';
import LogoutComponent from "./auth/LogoutComponent";
import ErrorComponent from "./auth/ErrorComponent";
import RegisterComponent from "./auth/RegisterComponent";
import WelcomeComponent from "./WelcomeComponent";
import ChatComponent from "./chat/ChatComponent"
import MeetupsComponent from "./meetup/MeetupsComponent";
import HomeComponent from "./HomeComponent";
import MeetupPageComponent from "./meetup/MeetupPageComponent"
import FriendsComponent from './invite/FriendsComponent';
import MeetupEventForm from "./meetup/MeetupEventForm"
import AuthenticatedRoute from "../accounts/AuthenticatedRoute";
import LoginComponent from "./auth/LoginComponent";
import Invites from "./invite/Invites"
import MeetupForm from "./meetup/MeetupForm"
import Profile from "./Profile"

class Body extends Component {

    render () {
        return (
            <div className="c">
                <Switch>
                    <Route path="/" exact component={HomeComponent}></Route>
                    <Route path="/login" component={LoginComponent}></Route>
                    <Route path="/register" component={RegisterComponent}></Route>
                    <AuthenticatedRoute path="/logout" component={LogoutComponent}></AuthenticatedRoute>
                    <AuthenticatedRoute path="/welcome" component={WelcomeComponent}></AuthenticatedRoute>
                    <AuthenticatedRoute path="/meetups/:uri/new" exact component={MeetupEventForm}></AuthenticatedRoute>
                    <AuthenticatedRoute path="/meetups/new" exact component={MeetupForm}></AuthenticatedRoute>
                    <AuthenticatedRoute path="/meetups/:uri" component={MeetupPageComponent}></AuthenticatedRoute>
                    <AuthenticatedRoute path="/meetups" exact component={MeetupsComponent}></AuthenticatedRoute>
                    <AuthenticatedRoute path="/chat/:uri" component={ChatComponent}></AuthenticatedRoute>
                    <AuthenticatedRoute path="/chat" exact component={ChatComponent}></AuthenticatedRoute>
                    <AuthenticatedRoute path="/friends" component={FriendsComponent}></AuthenticatedRoute>
                    <AuthenticatedRoute path="/invites" component={Invites}></AuthenticatedRoute>
                    <AuthenticatedRoute path="/profile/:id" component={Profile}></AuthenticatedRoute>
                    <Route component={ErrorComponent}></Route>
                </Switch>
            </div>
        )
    }
}

export default Body