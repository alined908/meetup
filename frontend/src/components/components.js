//Auth
import ErrorComponent from "./auth/ErrorComponent";
import LoginComponent from "./auth/LoginComponent";
import LogoutComponent from "./auth/LogoutComponent";
import RegisterComponent from "./auth/RegisterComponent";

//Chat
import ChatBarComponent from "./chat/ChatBarComponent"
import ChatComponent from "./chat/ChatComponent"
import ChatMessageComponent from "./chat/ChatMessageComponent"
import ChatWindowComponent from "./chat/ChatWindowComponent"
import ContactComponent from "./chat/ContactComponent"

//invite
import Friend from './invite/Friend';
import FriendsComponent from './invite/FriendsComponent';
import Invite from "./invite/Invite"
import Invites from "./invite/Invites"

//meetup
import Map from "./meetup/Map"
import Meetup from "./meetup/Meetup";
import MeetupCard from "./meetup/MeetupCard";
import MeetupEvent from "./meetup/MeetupEvent";
import MeetupEventForm from "./meetup/MeetupEventForm"
import MeetupForm from "./meetup/MeetupForm";
import MeetupFriend from "./meetup/MeetupFriend";
import MeetupPageComponent from "./meetup/MeetupPageComponent"
import MeetupsComponent from "./meetup/MeetupsComponent";
import Restauraunt from "./meetup/Restauraunt";

//fields
import CategoryAutocomplete from "./fields/CategoryAutocomplete"
import Location from "./fields/Location"
import renderTextField from "./fields/renderTextField"
import renderDatePicker from "./fields/renderDatePicker"
import renderDateSimplePicker from "./fields/renderDateSimplePicker"
import UserAutocomplete from "./fields/UserAutocomplete"

//General
import Body from "./general/Body"
import GlobalMessage from "./general/GlobalMessage"
import HomeComponent from "./general/HomeComponent";
import LiveUpdatingBadge from "./general/LiveUpdatingBadge"
import Navigation from "./general/Navigation"
import Profile from "./general/Profile"
import GroupAvatars from './general/GroupAvatars'
import NotificationWrapper from "./general/NotificationWrapper"
import CalendarComponent from "./general/CalendarComponent"
import CategoriesComponent from "./general/CategoriesComponent"
import CategoryComponent from "./general/CategoryComponent"
import Preferences from "./general/Preferences"
import Preference from "./general/Preference"
import SettingsComponent from "./general/SettingsComponent"

export {
    ErrorComponent, LoginComponent, LogoutComponent, RegisterComponent, 
    ChatBarComponent, ChatComponent, ChatMessageComponent, ChatWindowComponent, ContactComponent,
    Friend, FriendsComponent, Invite, Invites,
    Map, Meetup, MeetupCard, MeetupEvent, MeetupEventForm, MeetupFriend, MeetupForm, MeetupPageComponent, MeetupsComponent, Restauraunt,
    CategoryAutocomplete, UserAutocomplete, Location, renderTextField, renderDatePicker, renderDateSimplePicker,
    Body, GlobalMessage, GroupAvatars, HomeComponent, LiveUpdatingBadge, Navigation, Profile, NotificationWrapper, 
    Preferences, Preference, CalendarComponent, CategoriesComponent, CategoryComponent, SettingsComponent}