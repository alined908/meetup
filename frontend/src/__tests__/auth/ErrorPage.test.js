import React from "react";
import { shallow } from "enzyme";
import { ErrorPage } from "../../components/components";

describe("ErrorPage render", () => {
  it("renders without crashing", () => {
    const wrapper = shallow(<ErrorPage />);
    expect(wrapper).toMatchSnapshot();
  });
});
