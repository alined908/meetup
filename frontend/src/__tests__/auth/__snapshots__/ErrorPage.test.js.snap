// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage render renders without crashing 1`] = `
<div
  className="errorPage"
>
  <div
    className="errorPageIcon"
  >
    <Component
      height="100%"
      width="100%"
    />
  </div>
  <div
    className="errorPageMessage"
  >
    <div>
      Error code: 404
    </div>
    <div>
      We weren't able to find the page!
    </div>
    <div
      onClick={[Function]}
    >
      Try Again
    </div>
  </div>
</div>
`;
